packages/alchemy/src/provider/light-account.ts

```diff
@@ -0,0 +1,25 @@
+import {
+  getDefaultLightAccountFactoryAddress,
+  LightSmartContractAccount,
+} from "@alchemy/aa-accounts";
+import { LightAccountAlchemyProviderConfigSchema } from "../schema.js";
+import type { LightAccountAlchemyProviderConfig } from "../type.js";
+import { AlchemyProvider } from "./base.js";
+
+export const createLightAccountAlchemyProvider = (
```

Comment:
read this again, I'm curious there's a less verbose, more ergonomic approach to this factory.

for instance, what if we had an enum as an input that tells the factory what account to connect?

```
enum AccountType {
  SIMPLE_ACCOUNT,
  LIGHT_ACCOUNT
}

const createAlchemyProvider = ({...config, accountType: AccountType}) => {
  const provider = new AlchemyProvider(config)

  switch accountType: 
    case SIMPLE_ACCOUNT:
      provider.connect((rpcClient) => new SimpleSmartContractAccount({rpcClient, ...config})
    case LIGHT_ACCOUNT:
      provider.connect((rpcClient) => new LightSmartContractAccount({rpcClient, ...config})

  return provider
}

```